Try and fix youtube

mattheu opened the pull request:
FB doesn't seem to like our markup. There are a few differences I've noticed from ours, the documentation, and other feeds
1. Our embeds are wrapped in a div or span with the class like `embed-`. We can fairly easily unwrap these using.
2. We're left with something like the following. `<iframe><iframe src="">`. I don't think this is 100% necessary,  and the video will render full width if we can strip the outer iframe. However we can only do this if the embed uses an iframe, and not if using scripts and other markup, so I'm quite strict about checking this is the only element.

mattheu committed:
Try and fix some bad markup in embeds

mattheu committed:
Missing }

mattheu committed:
cleanup old iframe

dashaluna commented on simple-facebook-instant-articles.php:
```diff
@@ -399,6 +401,49 @@ public function reformat_social_embed( $html, $url, $attr, $post_ID = null ) {
+				$node->parentNode->removeChild( $node );
+			}
+
+			while ( $node->childNodes->length > 0 ) {
```
Would be good to have a comment for quick tip what we're doing here.

dashaluna committed:
Add comments for clarity

dashaluna committed:
Added comments for clarity

dashaluna commented on simple-facebook-instant-articles.php:
```diff
@@ -402,6 +403,55 @@ public function reformat_social_embed( $html, $url, $attr, $post_ID = null ) {
+
+		// Matches all divs and spans that have class like ~=embed- and are children of figure.
+		// Unwrap, or remove if no children.
+		$items = $xpath->query( '//figure[contains(@class, \'op-social\')]//*[self::span or self::div][contains(@class, \'embed-\')]' );
```
@mattheu did you mean `span` and `div` are children of figure or descendants?

mattheu replied:
Yep I do mean decendants ;)

dashaluna commented:
I know ;) but down the line say 6 months, it's just clearer to see what we're doing and that `$node` is either `div` or `span` rather than going back to the xpath.

dashaluna committed:
Edit comment to reflect xpath

mattheu commented:
OK - I think we can reuse the code for `unwrap` so i've moved to a separate method and added better documentation. I think its much neater now. Thanks ;)

mattheu commented:
Oh oops you merged ;) Was still fixing your comments ;)
